packages/hooks/src/index.js

```diff
@@ -27,8 +27,8 @@ export const doAction     = createRunHook( HOOKS.actions );
 export const applyFilters = createRunHook( HOOKS.filters, true );
 
 // Current action/filter functions.
-export const currentAction = () => HOOKS.actions.current || null;
-export const currentFilter = () => HOOKS.filters.current || null;
+export const currentAction = () => HOOKS.actions.__current[0] || null;
+export const currentFilter = () => HOOKS.filters.__current[0] || null;
```

Comment:
This should return the last element instead of the first, and it needs unit tests for multiple filters running at the same time.  It's probably worth splitting it out into a separate file again.